src/btree/bt_cursor.c

```diff
@@ -1051,17 +1058,11 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
         exact = cbt->compare;
```

Comment:
You could simplify this block a lot given that if bounds_resosition_exact is non zero we know bounds_set is true.
```
    if (valid) {
        /* If the bounded cursor logic respositioned the cursor override exact. */
        exact = bounds_reposition_exact != 0 ? bounds_reposition_exact : cbt->compare;
        WT_ERR(__cursor_kv_return(cbt, cbt->upd_value));
```

Reply:
[47d8230](https://github.com/wiredtiger/wiredtiger/pull/9914/commits/47d82300ff604405549a85dc106dcc0c84cf4bd7)